api/v1alpha4/vspheremachine_webhook.go

```diff
@@ -59,6 +59,8 @@ func (m *VSphereMachine) ValidateCreate() error {
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (m *VSphereMachine) ValidateUpdate(old runtime.Object) error {
 	newVSphereMachine, err := runtime.DefaultUnstructuredConverter.ToUnstructured(m)
+	spec := m.Spec
```

Comment:
nit: move this closer to the for loop and maybe rename? Gets a bit confusing with multiple machine specs. There's already a oldVSphereMachineSpec so name would be difficult :p

Reply:
would newMachineSpec be a more speaking name?
Ive kept it at spec as that name and logic got already used in the creation validation

Reply:
yah maybe just leave it as spec. moving it closer to the for loop should make it clearer

Reply:
Ok, done.Ive also added a comment to clarify hat the block does

Reply:
thanks, squash the commits and i can lgtm

Reply:
and squashed